### PR TITLE
Add docs for egressMasqueradeInterfaces option in Cilium

### PR DESCRIPTION
**Pull request by ahreehong**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/getting-started/optional/cni.md [hunk 1]
```diff
@@ -137,6 +137,33 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
 will not delete any of the existing NetworkPolicy objects, including the ones required
    for EKS Anywhere components (listed above). The user must delete NetworkPolicy objects as needed.
 
+### EgressMasqueradeInterfaces option for Cilium plugin
+
+Cilium accepts the EgressMasqueradeInterfaces option from the users to limit which interfaces masquerading is performed on.
+The allowed values for this mode are an `interface name` or an `interface prefix`.
+Please refer the official [Cilium documentation](https://docs.cilium.io/en/v1.13/network/concepts/masquerading/#iptables-based) for more details on how this option affects masquerading traffic.
+
+By default, masquerading will be performed on all traffic leaving on a non-Cilium network device. This only has an effect on traffic egressing from a node to an external destination not part of the cluster and does not affect routing decisions. In some cases, it also will be required to configure the [ipv4 native routing cidr](https://docs.cilium.io/en/v1.13/network/concepts/masquerading/#configuration).
```
Comment (jacobweinstock) [inline docs/content/en/docs/getting-started/optional/cni.md:146]: `In some cases, it also will be required to configure the [ipv4 native routing cidr](https://docs.cilium.io/en/v1.13/network/concepts/masquerading/#configuration)` I would recommend not including this. it opens up a can of worms that we would need to explain. If you do feel we should keep this, then i would suggest being specific about the cases. if i read this as a customer how do i know if my case warrants configuring native routing.

**Review by jacobweinstock — CHANGES_REQUESTED** [review]

File: docs/content/en/docs/getting-started/optional/cni.md [hunk 2]
```diff
@@ -137,6 +137,33 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
 will not delete any of the existing NetworkPolicy objects, including the ones required
    for EKS Anywhere components (listed above). The user must delete NetworkPolicy objects as needed.
 
+### EgressMasqueradeInterfaces option for Cilium plugin
```
Comment (jacobweinstock) [inline docs/content/en/docs/getting-started/optional/cni.md:140]: plugin? do we use this term other places? if not, i would suggest using maybe use `option for the Cilium CNI` or similar.

File: docs/content/en/docs/getting-started/optional/cni.md [hunk 3]
```diff
@@ -137,6 +137,33 @@ EKS Anywhere will create the required NetworkPolicy objects for its core compone
...
+
+Cilium accepts the `EgressMasqueradeInterfaces` option from users to limit which interfaces masquerading is performed on.
+The allowed values for this mode are an `interface name` such as `eth0` or an `interface prefix` such as `eth+`.
+Please refer the official [Cilium documentation](https://docs.cilium.io/en/v1.13/network/concepts/masquerading/#iptables-based) for more details on how this option affects masquerading traffic.
```
Comment (chrisnegus) [inline docs/content/en/docs/getting-started/optional/cni.md:144]:
```suggestion
Please refer to the official [Cilium documentation](https://docs.cilium.io/en/v1.13/network/concepts/masquerading/#iptables-based) for more details on how this option affects masquerading traffic.
```

File: docs/content/en/docs/getting-started/optional/cni.md (hunk 2 above)
Comment (ahreehong) [reply docs/content/en/docs/getting-started/optional/cni.md:140]: We use plugin here: `Policy Configuration options for Cilium plugin`

**Review by jacobweinstock — APPROVED** [review]

**Review by jacobweinstock — APPROVED** [review]

**Pull request merged**